diff --git a/src/sbattach.c b/src/sbattach.c
index a0c01b8..4a0eea9 100644
--- a/src/sbattach.c
+++ b/src/sbattach.c
@@ -148,12 +148,6 @@ static int remove_sig(struct image *image, int signum,
 
 	rc = image_remove_signature(image, signum);
 
-	if (rc) {
-		fprintf(stderr, "Error, image has no signature at %d\n",
-			signum + 1);
-		return rc;
-	}
-
 	rc = image_write(image, image_filename);
 	if (rc)
 		fprintf(stderr, "Error writing %s: %s\n", image_filename,
